### DISCUSSION TITLE
I edited a ranked beatmap and accidentally saved it, how do I revert back safely?

### DISCUSSION THREAD

**Discussion by JacksonChen666** [ppy/osu#24047] [Q&A] [answered]
~~Currently (as of 2023-06-27T11:43:11+02:00), I have osu!lazer open (I also sent a SIGSTOP to suspend the process for now).~~ I also made a copy of my osu folder after editing but before closing osu!lazer so I should still have the files.

I edited a ranked beatmap, accidentally saved it and my scores for the difficulty is now gone. I want to revert back to before I saved the beatmap (keep my local scores and stuff). How could I do that?

**Comment by Theighlin** [comment]
Your local scores for the single difficulty(ies) you modified are gone. Updating the beatmap is not working currently  and being tracked here #23524.

**Reply by peppy** [reply]
The scores shouldn't be gone. They will return when you re-download the beatmap or update it back to the ranked version.

**Reply by Theighlin** [reply]
They are definitely gone, i said that without trying, but i did just now and happens both with.

1) Scores on ranked beatmap -> save modified beatmap-> delete beatmap-> redownload without closing game -> scores are not there.
2) Scores on ranked beatmap -> save modified beatmap -> delete beatmap-> redownload after game restart -> scores are not there.

Note that you can't update the beatmap as of now, but i what happens in the back is the same as redownloading i take it.

I did ask you on stream about this once and you said keeping scores didn't make sense or something like that?
Keeping scores of the original unedited beatmap sounds good though.

**Reply by peppy** [reply]
I guess they aren't gone if you undelete the updated beatmap before restarting.

**Reply by JacksonChen666** [reply]
scores on ranked beatmap -> make a modification and save it -> download beatmap on web and import it without first deleting it -> restore deleted beatmaps -> scores are not there, not even on other difficulties on the new map

even after restart, scores are still not there.

**Reply by peppy** [reply]
Coming back to this, scores are actually not deleted anywhere in code.

So this probably could have been recovered by relinking them in the database.

Not sure how easy it will be to make this happen automatically, but it could happen.

**Comment by JacksonChen666** [comment, marked as answer]
I've resorted to using a backup of my osu folder (from my system backups) instead as it seems like there isn't a way back.

I used a backup that's about 2 hours before I created this discussion, so I completely lost whatever was between the 2 hours in the map that I edited (which is probably only [a single score](https://lazer.ppy.sh/scores/2346498722)).
